module/VuFind/src/VuFind/Db/Service/CommentsService.php

```diff
@@ -39,7 +39,100 @@
  * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
  */
 class CommentsService extends AbstractService
+    implements \VuFind\Db\Service\ServiceAwareInterface
 {
+    use \VuFind\Db\Service\ServiceAwareTrait;
+
+    /**
+     * Create a comments entity object.
+     *
+     * @return Comments
+     */
+    public function createEntity(): Comments
+    {
+        $class = $this->getEntityClass(Comments::class);
+        return new $class;
+    }
+
+    /**
+     * Get tags associated with the specified resource.
+     *
+     * @param string $id     Record ID to look up
+     * @param string $source Source of record to look up
+     *
+     * @return array
+     */
+    public function getForResource(string $id, $source = DEFAULT_SEARCH_BACKEND)
+    {
+        $resource = $this->getDbService(\VuFind\Db\Service\ResourceService::class)
+            ->findResource($id, $source);
+
+        if (empty($resource)) {
+            return [];
+        }
+        $dql = "SELECT c, u.firstname, u.lastname "
+            . "FROM " . $this->getEntityClass(Comments::class) . " c "
+            . "LEFT JOIN c.user u "
+            . "WHERE c.resource = :resource "
+            . "ORDER BY c.created";
+
+        $parameters['resource'] = $resource;
```

Comment:
`$parameters = compact('resource');` might be a better way to write this (otherwise in strict mode PHP might complain about writing into an uninitialized array).